### ISSUE TITLE
TextEdit with autowrap and fit content height does not work correctly

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#81764]
### Godot version

4.2 dev4

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

When you set TextEdit's wrap mode to boundary and enable `scroll_fit_content_height`, the TextEdit is expected to never show scrollbars. But it does show scrollbars at certain sizes, both horizontal and vertical.

https://github.com/godotengine/godot/assets/2223172/375b6b76-0e5b-4233-ac96-b187ed10686a

I remember this worked seamlessly when the fit content feature was first added.

Moreover, if the text is sufficiently long, it can freeze at certain sizes.

https://github.com/godotengine/godot/assets/2223172/8e6817c5-d1d8-431d-8102-bafc4bf7e410

(also notice how horizontal scroll appears despite using autowrap)

EDIT:
Seems like https://github.com/godotengine/godot/issues/80546 covers the freeze (crash) part. But the scrollbar bug is still relevant here.

### Steps to reproduce

1. Add TextEdit
2. Set `wrap_mode` to Boundary
3. Enable `scroll_fit_content_height`
4. Put any text longer than ~10 words
5. Resize

### Minimal reproduction project

N/A

**Comment by bitsawer** [comment]
Looks like a duplicate of https://github.com/godotengine/godot/issues/80546

**Comment by KoBeWi** [comment]
I edited the issue, it's only a partial duplicate.

**Comment by matheusmdx** [comment]
Bisecting shows that issue started after #74813
